### DISCUSSION TITLE
Rendering in ContinuousSpace

### DISCUSSION THREAD

**Discussion by mohitmasiwal95** [mesa/mesa#2094] [Q&A]
Hello, I am trying to simulate passenger flow through a space, but I am having trouble rendering the model because, I believe, ContinuousSpace lacks the **get_cell_list_contents** attribute, which is required for rendering. Could someone please assist me?

**Comment by EwoutH** [comment]
Hi, thanks for reaching out!

@rht I think you are most up to speed with our visualisation, could you help this user out?

**Reply by rht** [reply]
@awtishtikh you might want to check out the visualization of Boid Flocker (which uses continuous space) at https://github.com/projectmesa/mesa-examples/blob/main/examples/boid_flockers/app.py, and the code implementation of how it is done at https://github.com/projectmesa/mesa/blob/51b4e9a4c98c9e242cbe7bbf3b61275719a3126a/mesa/experimental/components/matplotlib.py#L76-L115. It's only a few lines of code.

**Reply by rht** [reply]
This is an example of how to create your own custom space drawer: https://github.com/projectmesa/mesa-examples/blob/cd09aba92b1f63a98fa4400c00d93d3fc1dd5c19/examples/sugarscape_g1mt/app.py#L9-L45
